src/metadata/audiotagger.cpp

```diff
@@ -0,0 +1,107 @@
+#include "metadata/audiotagger.h"
+
+#include "metadata/trackmetadatataglib.h"
+
+#include "util/assert.h"
+
+#include <taglib/vorbisfile.h>
+#include <taglib/wavpackfile.h>
+#include <taglib/mpegfile.h>
+#include <taglib/mp4file.h>
+#include <taglib/oggfile.h>
+#include <taglib/flacfile.h>
+#include <taglib/aifffile.h>
+#include <taglib/rifffile.h>
+#include <taglib/wavfile.h>
+#if (TAGLIB_MAJOR_VERSION > 1) || ((TAGLIB_MAJOR_VERSION == 1) && (TAGLIB_MINOR_VERSION >= 9))
+#include <taglib/opusfile.h>
+#endif
+
+#include <QtDebug>
+
+namespace {
+
+inline SecurityTokenPointer openSecurityToken(QFileInfo file,
+        const SecurityTokenPointer& pToken) {
+    if (pToken.isNull()) {
+        return Sandbox::openSecurityToken(file, true);
+    } else {
+        return pToken;
+    }
+}
+
+} // anonymous namespace
+
+AudioTagger::AudioTagger(const QString& file, SecurityTokenPointer pToken) :
+        m_file(file),
+        m_pSecurityToken(openSecurityToken(m_file, pToken)) {
+}
+
+AudioTagger::~AudioTagger() {
+}
+
+bool AudioTagger::save(const Mixxx::TrackMetadata& trackMetadata) {
```

Comment:
Btw, reading of audio data and the reading/writing of metadata should be separated. For audio data we now have AudioSource and for metdata we need some kind of MetadataSource. The SoundSource in the middle will become obsolete eventually.

Currently there is a lot of code duplication in parseHeader()/parseCover() that is implemented independently by every SoundSourceXXX. A generic function for all known file types is needed anyway -> see pull request for FFmpeg using TagLib for reading metadata.